File: java/Home_tasks/Car fuel system/CarFuelSystem.java
public class CarFuelSystem {
    private boolean carState = false;
    private boolean movingState = false;
    private int maximumFuelCapacity = 50;
    private int currentFuelLevel = 0;
    private int fuelEfficiency = 10;

    public int getFuelTankLevel() {
        return currentFuelLevel;
    }

    public int addFuel(int fuelAmount) {
        if (fuelAmount < 0) {
            return currentFuelLevel;
        }
        currentFuelLevel += fuelAmount;
        if (currentFuelLevel > maximumFuelCapacity) {
            currentFuelLevel = maximumFuelCapacity;
        }
        return currentFuelLevel;
    }


    public void startCar() {
        this.carState = true;
    }

    public boolean stopCar() {
        this.carState = false;
        return carState;
    }   


    public void driveCar(int distance) {
        if(!carState) {
            System.out.println("Start the car first.");
        }
        int fuelNeeded = distance / fuelEfficiency;
        if (currentFuelLevel >= fuelNeeded) {
            currentFuelLevel -= fuelNeeded;
            this.movingState = true;
        } 
        
        else {
            System.out.println("Not enough fuel to drive the desired distance.");
        }
    }
}

File: JS/Home_Tasks/CarFuelSystem.js
let carState = false;
let fuelEfficiency = 10;
let movingState = false;
let maxFuelCapacity = 50;       
let currentFuelLevel = 0;

function getFuelTankLevel() {
    return currentFuelLevel;
}

function addFuel(fuel) {
    if (fuel <= 0) {
        return 0;
    }
    if (currentFuelLevel + fuel > maxFuelCapacity) {
        currentFuelLevel = maxFuelCapacity;
        return currentFuelLevel;
    } else {
        currentFuelLevel += fuel;
        return currentFuelLevel;
    }
}

function startCar() {
    carState = true;
}   

function driveCar(distance) {
    if(!carState) {
        console.log("Start the car first.");
        return false;
    }
    let fuelNeeded = distance / fuelEfficiency;
    if (currentFuelLevel >= fuelNeeded) {
        currentFuelLevel -= fuelNeeded;
        movingState = true;
        return true;
    } else {
        console.log("Not enough fuel to drive the distance.");
        return false;
    }   

}
function stopCar() {
    carState = false;
    movingState = false;
    return carState;
}       

module.exports = {
    getFuelTankLevel,
    addFuel,    
    startCar,
    driveCar,
    stopCar
};


